### ISSUE TITLE
Several parallel connections praxis

### ISSUE THREAD

**Issue by dargolith** [mogui/pyorient#145]
This is a question (or 2 actually).

I need to run queries in parallel from python. The idea is to use several instances of pyorient.OrientDB, one for each parallel command.

Example:

    Incoming request -> query1 starts
    context switch
    incoming request -> query2 starts
    context switch

Of course, one object won't do since the socket will already be used when query2 attempts to use it. So my idea was to make a pool of pyorient.OrientDB objects. The database to perform actions on can be different for each request / query.

I was then starting to wonder about open sockets and so on and what would be the most efficient way. Normally I guess you want to open a socket to the db, perform the command, close it. Repeating for each request. But I wasn't sure how pyorient was meant to deal with this. So I started investigating a bit what happens underneath.

To me it seems that the underlying socket is being opened as the message is created (init in BaseMessage runs get_connection() on the OrientSocket). However, I could never find this socket closing, except when pyorient.OrientDB.db_close() is being called, resulting in the BaseMessage.close() eventually being run, which will close the OrientSocket. However, at that point the OrientDB-object is left in a state that I don't quite understand.

If I run:

    db = pyorient.OrientDB('localhost', 2424)
    db.db_open('testdb', 'user', 'password')
    db.db_close()

Everything works fine and I guess the socket is being closed, but any attempt to perform any action (db_open(), connect(), or anything really) just throws PyOrientConnectionExceptions, saying: "SocketError: [Error 10049] The requested address is not valid in its context".

Question1:
What is the OrientDB object state after db_close() has been called? (can it be further used at all?)

Question2:
What would be most efficient regarding to a high amount of request resulting in several queries to different orient databases? (New OrientDB object for each request? Not closing them and just db_open() again?)

**Comment by Ostico** [comment]
Hi @dargolith ,
regarding the first question, after db_close the client can not be used anymore and it must be re-initialized:
```
db = pyorient.OrientDB('localhost', 2424)
```
Regarding the second question i suggest you to make a pool of clients ( one for physical server ) and issue a db_open command everytime you must change the database without close them.

Otherwise you can create a pool of clients, one for database on which the client must operate and switch over them programmatically.

Let me know if you need more info.

**Comment by dargolith** [comment]
Hi, 

I already ended up doing what you suggested (first variant). I made a pool of connections, on server level (same connection pool for different databases), issuing db_open for each new request.

Thanks for answer!
